// Components/News/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'

import {requestNews} from 'Actions'

class News extends Component{
	constructor(props){
		super(props);
		this.state = {
			// loading: false,
			// loaded: false,
			// error: false
		}
	}
	componentDidMount(){
		this.props.requestNews('https://mysterious-reef-29460.herokuapp.com/api/v1/news')

		this.setState(() => {
			return {
				loading: true
			};
		})
	}

	render(){
		const getBody = () => {
			const loading = this.props
			const {data, error} = this.props.news
			if( loading){
				return(
					<p>Loading...</p>
				)
			}
			if( error){
				return(
					<p className="error">Error</p>
				)
			}
			if( data ){
				const {city, languages, social} = data
			}
		}
		return(
			<div className="container">
				{getBody()}
			</div>
		)
	}
}

const mapStateToProps = (state) => ({
	news: state.news
})

const mapDispatchToProps = (dispatch) => ({
	requestNews: (link) => dispatch(requestNews(link))
})

export default connect(mapStateToProps, mapDispatchToProps)(News)
