// Components/News/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import PropTypes from 'prop-types'

import {requestNews} from 'Actions'

import './style.css'

class News extends Component{
	componentDidMount(){
		if( !this.props.news.loaded && !this.props.news.loading)
			this.props.requestNews('https://mysterious-reef-29460.herokuapp.com/api/v1/news')
	}

	render(){
		const getBody = () => {
			const {loading, data, error} = this.props.news
			if( loading){
				return(
					<i className="fa fa-spinner fa-spin"></i>
				)
			}
			if( error){
				return(
					<p className="error">Error</p>
				)
			}
			if( data ){
				return <div className="news">
					<div>
						{getNews(data)}
					</div>
					<div>
						{mountNews(data)}
					</div>
				</div>
			}
			function getNews(data){
				return data.map( news => {
					return <div className="b-news" key={news.id}>
						<div className="b-news__title">{news.title}</div>
						<div className="b-news__txt">{news.text}</div>
					</div>
				})
			}
			function mountNews(data){
				return <div className="b-news__mount"><b>всего новостей:</b> {data.length}</div>
			}
		}
		return(
			<div className="container">
				{getBody()}
			</div>
		)
	}
}

News.propTypes = {
	news: PropTypes.shape({
		loading: PropTypes.bool.isRequired,
		loaded: PropTypes.bool.isRequired,
		data: PropTypes.array.isRequired
	})
}

const mapStateToProps = (state) => ({
	news: state.news
})

const mapDispatchToProps = (dispatch) => ({
	requestNews: (link) => dispatch(requestNews(link))
})

export default connect(mapStateToProps, mapDispatchToProps)(News)
